clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -415,6 +439,27 @@ public ConsumerRecords<K, V> poll(final Duration timeout) {
             return ConsumerRecords.empty();
         } finally {
             kafkaConsumerMetrics.recordPollEnd(timer.currentTimeMs());
+            wakeupTrigger.clearActiveTask();
```

Comment:
Hmm, I'm not 100% sure we are providing the guarantee stated in the javadoc of the consumer here. 

This would be: 

```
     * If no thread is blocking in a method which can throw {@link org.apache.kafka.common.errors.WakeupException}, the next call to such a method will raise it instead.
```

But if the current fetch returns records, we will never throw a `WakeupException`, not from this `poll` nor the next. Should we keep the "activeTask" around strictly until we are ready to throw a `WakeupException`?

Reply:
I think you are right @lucasbru - We need to ensure WakeupException will also be thrown when wakeup() is invoked.   I wonder if we need to wrap the whole poll() in the future.

Reply:
I removed it. We no longer need it.